code/game/objects/items/grenades/atmos_grenades.dm

```diff
@@ -97,3 +97,38 @@
 		var/turf/open/floor_loc = turf_loc
 		floor_loc.atmos_spawn_air("n2o=[n2o_gas_amount / distance_from_center];TEMP=273")
 	qdel(src)
+
+/obj/item/grenade/gas_crystal/crystal_foam
+	name = "crystal foam"
+	desc = "A crystal with a foggy inside"
+	icon_state = "crystal_foam"
+	var/list/obj/item/reagent_containers/glass/beakers = list()
+	var/breach_range = 7
+
+/obj/item/grenade/gas_crystal/crystal_foam/detonate(mob/living/lanced_by)
+	. = ..()
+
+	var/list/reagent_list = list()
+	var/datum/reagents/first_batch = new
+	var/datum/reagents/second_batch = new
+
+	first_batch.add_reagent(/datum/reagent/aluminium, 75)
+	second_batch.add_reagent(/datum/reagent/smart_foaming_agent, 25)
+	second_batch.add_reagent(/datum/reagent/toxin/acid/fluacid, 25)
+	reagent_list += first_batch
+	reagent_list += second_batch
+
+	var/list/datum/reagents/reactants = list()
+	for(var/batch in reagent_list)
+		reactants += batch
```

Comment:
Why add them to a list twice?

Reply:
Wasn't sure if was possible, I'll try to directly add them to one list